crates/nu-command/src/debug/info.rs

```diff
@@ -0,0 +1,136 @@
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    record, Category, Example, IntoPipelineData, PipelineData, Record, ShellError, Signature, Span,
+    Type, Value,
+};
+use sysinfo::{Pid, PidExt, ProcessExt, ProcessRefreshKind, RefreshKind, System, SystemExt};
+
+#[derive(Clone)]
+pub struct DebugInfo;
+
+impl Command for DebugInfo {
+    fn name(&self) -> &str {
+        "debug info"
+    }
+
+    fn usage(&self) -> &str {
+        "View process memory info."
+    }
+
+    fn extra_usage(&self) -> &str {
+        "This command is meant for debugging purposes.\nIt shows you the process information and system memory information."
+    }
+
+    fn signature(&self) -> nu_protocol::Signature {
+        Signature::build("debug info")
+            .input_output_types(vec![(Type::Nothing, Type::Record(vec![]))])
```

Comment:
why not specifying the type a bit more? :yum: 
```suggestion
            .input_output_types(vec![(
                Type::Nothing,
                Type::Record(vec![
                    ("pid".into(), Type::Int),
                    ("ppid".into(), Type::Int),
                    (
                        "process".into(),
                        Type::Record(vec![
                            ("memory".into(), Type::Filesize),
                            ("virtual_memory".into(), Type::Filesize),
                            ("status".into(), Type::String),
                            ("root".into(), Type::String),
                            ("cwd".into(), Type::String),
                            ("exe_path".into(), Type::String),
                            ("command".into(), Type::String),
                            ("name".into(), Type::String),
                            ("environment".into(), Type::Record(vec![])),
                        ]),
                    ),
                    (
                        "system".into(),
                        Type::Record(vec![
                            ("total_memory".into(), Type::Filesize),
                            ("free_memory".into(), Type::Filesize),
                            ("used_memory".into(), Type::Filesize),
                            ("available_memory".into(), Type::Filesize),
                        ]),
                    ),
                ]),
            )])
```
which gives the following input / output signature in `help debug info` :smirk: 
```
Input/output types:
  ╭───┬─────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
  │ # │  input  │                                                    output                                                    │
  ├───┼─────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ 0 │ nothing │ record<pid: int, ppid: int, process: record<memory: filesize, virtual_memory: filesize, status: string,      │
  │   │         │ root: string, cwd: string, exe_path: string, command: string, name: string, environment: record>, system:    │
  │   │         │ record<total_memory: filesize, free_memory: filesize, used_memory: filesize, available_memory: filesize>>    │
  ╰───┴─────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

> **Note**
> because the shape of `$.process.environment` is unknown, its type is just `record<>`

Reply:
I'm fine with it being just record because that leaves room for this to mature and change more easily.

Reply:
we have a bunch of underspecified signatures, i thought this could be a start to show how to do it properly :relieved: 

if a few fields of the output record change in a followup PR, i would say it's the responsibility of the author to change it accordingly :yum:

Reply:
I'm not really a fan of having an extra 30 lines of boiler plate just to specific an output type. I don't really see it as useful. Maybe I'm not understanding the value.

Reply:
> I'm not really a fan of having an extra 30 lines of boiler plate just to specific an output type. I don't really see it as useful. Maybe I'm not understanding the value.
>
we have a type system that can check and enforce types of the values one manipulates in pipelines, i think it's useful to specify types as much as we can when we know them in advance :yum: 
otherwise, i just feel we do not leverage the full potential of the type system of Nushell :cry:

Reply:
just leaving this open for future us to see it if need be :relieved: